examples/qml_minimal/CMakeLists.txt

```diff
@@ -22,125 +22,208 @@ set(CMAKE_AUTOMOC ON)
-# to set the CXXQT_EXPORT_DIR environment variable when calling `cargo build`.
-# Also, set the QMAKE environment variable to ensure the Rust library uses
-# the same installation of Qt as CMake.
-set(CXXQT_EXPORT_DIR "${CMAKE_CURRENT_BINARY_DIR}/cxxqt")
-corrosion_set_env_vars(${CRATE}
-    "CXXQT_EXPORT_DIR=${CXXQT_EXPORT_DIR}"
-    "QMAKE=${QMAKE}"
-    $<$<BOOL:${CMAKE_RUSTC_WRAPPER}>:RUSTC_WRAPPER=${CMAKE_RUSTC_WRAPPER}>
-)
-
-# Create an INTERFACE library target to link libraries to and add include paths.
-# Linking this to both the application and the tests avoids having to setup
-# the include paths and linked libraries for both of those.
-add_library(${APP_NAME}_lib INTERFACE)
-
-# Include the headers generated by the Rust library's build script. Each
-# crate gets its own subdirectory under CXXQT_EXPORT_DIR. This allows you
-# to include headers generated by multiple crates without risk of one crate
-# overwriting another's files.
-target_include_directories(${APP_NAME}_lib INTERFACE "${CXXQT_EXPORT_DIR}/${CRATE}")
-
-target_link_libraries(${APP_NAME}_lib INTERFACE
-    # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
-    # otherwise the linker will discard the static variables that initialize it.
-    "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
-    Qt::Core
-    Qt::Gui
-    Qt::Qml
-    Qt::QuickControls2
-)
-# ANCHOR_END: book_cmake_use_corrosion
-
-# ANCHOR: book_cmake_executable
-# Define the executable with the C++ source
-add_executable(${APP_NAME} cpp/main.cpp)
-
-# Link to the Rust library
-target_link_libraries(${APP_NAME} PRIVATE ${APP_NAME}_lib)
-
-# If we are using a statically linked Qt then we need to import any qml plugins
-qt_import_qml_plugins(${APP_NAME})
-# ANCHOR_END: book_cmake_executable
+    # Create an INTERFACE library target to link libraries to and add include paths.
+    # Linking this to both the application and the tests avoids having to setup
+    # the include paths and linked libraries for both of those.
+    add_library(${APP_NAME}_lib INTERFACE)
+
+    # Include the headers generated by the Rust library's build script. Each
+    # crate gets its own subdirectory under CXXQT_EXPORT_DIR. This allows you
+    # to include headers generated by multiple crates without risk of one crate
+    # overwriting another's files.
+    target_include_directories(${APP_NAME}_lib INTERFACE "${CXXQT_EXPORT_DIR}/${CRATE}")
+
+    target_link_libraries(${APP_NAME}_lib INTERFACE
+        # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
+        # otherwise the linker will discard the static variables that initialize it.
+        Qt::Core
+        Qt::Gui
+        Qt::Qml
+        Qt::QuickControls2
+    )
+    # ANCHOR_END: book_cmake_use_corrosion
 
-#
-# Unit test
-#
+    target_link_options(${APP_NAME}_lib INTERFACE "SHELL:-Wl,--whole-archive $<TARGET_FILE:${CRATE}-static> -Wl,--no-whole-archive")
```

Comment:
generally instead of mass if blocks, we should try not to change this code too much as it's used in the books etc as well.

Instead can we have a minimal diff and have tight if blocks eg

```cmake
# WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
# otherwise the linker will discard the static variables that initialize it.
if(EMSCRIPTEN)
    target_link_libraries(${APP_NAME}_lib INTERFACE
        Qt::Core
        Qt::Gui
        Qt::Qml
        Qt::QuickControls2
    )
    target_link_options(${APP_NAME}_lib INTERFACE "SHELL:-Wl,--whole-archive $<TARGET_FILE:${CRATE}-static> -Wl,--no-whole-archive")
else()
    target_link_libraries(${APP_NAME}_lib INTERFACE
       "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
       Qt::Core
       Qt::Gui
       Qt::Qml
       Qt::QuickControls2
    )
endif()
```

Then it's much easier to see where the differences are for wasm.